nowcasting_dataset/data_sources/optical_flow/optical_flow_data_source.py

```diff
@@ -0,0 +1,267 @@
+        """
+        # Combine all channels for a single timestep
+        satellite_data = satellite_data.isel(
+            time_index=slice(
+                satellite_data.sizes["time_index"] - predictions.shape[0],
+                satellite_data.sizes["time_index"],
+            )
+        )
+        # Make sure its the correct size
+        buffer = (satellite_data.sizes["x_index"] - self.image_size_pixels) // 2
+        satellite_data = satellite_data.isel(
+            x_index=slice(buffer, satellite_data.sizes["x_index"] - buffer),
+            y_index=slice(buffer, satellite_data.sizes["y_index"] - buffer),
+        )
+        dataarray = xr.DataArray(
+            data=predictions,
+            dims={
+                "time_index": satellite_data.dims["time_index"],
+                "x_index": satellite_data.dims["x_index"],
+                "y_index": satellite_data.dims["y_index"],
+                "channels_index": satellite_data.dims["channels_index"],
+            },
+            coords={
+                "time_index": satellite_data.coords["time_index"],
+                "x_index": satellite_data.coords["x_index"],
+                "y_index": satellite_data.coords["y_index"],
+                "channels_index": satellite_data.coords["channels_index"],
+            },
+        )
+
+        return dataarray
+
+    def _get_previous_timesteps(
+        self,
+        satellite_data: xr.DataArray,
+        t0_dt: pd.Timestamp,
+    ) -> xr.DataArray:
+        """
+        Get timestamp of previous
+
+        Args:
+            satellite_data: Satellite data to use
+            t0_dt: Timestamp
+
+        Returns:
+            The previous timesteps
+        """
+        satellite_data = satellite_data.where(satellite_data.time <= t0_dt, drop=True)
+        return satellite_data
+
+    def _get_number_future_timesteps(
+        self, satellite_data: xr.DataArray, t0_dt: pd.Timestamp
+    ) -> int:
+        """
+        Get number of future timestamps
+
+        Args:
+            satellite_data: Satellite data to use
+            t0_dt: The timestamp of the t0 image
+
+        Returns:
+            The number of future timesteps
+        """
+        satellite_data = satellite_data.where(satellite_data.time > t0_dt, drop=True)
+        return len(satellite_data.coords["time_index"])
+
+    def _compute_and_return_optical_flow(
+        self,
+        satellite_data: xr.DataArray,
+        t0_dt: pd.Timestamp,
+    ) -> xr.DataArray:
+        """
+        Compute and return optical flow predictions for the example
+
+        Args:
+            satellite_data: Satellite DataArray
+            t0_dt: t0 timestamp
+
+        Returns:
+            The Tensor with the optical flow predictions for t0 to forecast horizon
+        """
+
+        # Get the previous timestamp
+        future_timesteps = self._get_number_future_timesteps(satellite_data, t0_dt)
+        satellite_data: xr.DataArray = self._get_previous_timesteps(
```

Comment:
Just to be really explicit, please consider renaming `satellite_data` to something like `historical_satellite_data`